libs/runtime/ebpf_hash_table.c

```diff
@@ -597,7 +629,7 @@ _ebpf_hash_table_replace_bucket(
     }
 
     // Find the old bucket.
-    old_bucket = hash_table->buckets[bucket_index].header;
+    old_bucket = _ebpf_hash_table_get_bucket(hash_table, bucket_index);
```

Comment:
question (for my understanding): against which instruction are we trying to prevent reordering?

Reply:
This has been moved to it's own PR:
https://github.com/microsoft/ebpf-for-windows/pull/3890

Short version:
Find path has two reads:
1) Read the pointer to the bucket from array.
2) Read the values from the bucket.

Update path has two writes:
1) Write value to bucket.
2) Write pointer to bucket into array.

We need to ensure that the order is correctly enforced. Not an issue on x64, but potentially an issue on ARM64.

Reply:
thanks. follow-up question (though this is a draft PR now): did we hit any re-ordering issue in our tests on ARM64?

Reply:
No, but this is based on advice from the advice from the ARM64 experts. To hit the suspected error we would need:
1) A hash-collision.
2) Concurrent read of the previous key in the bucket.
3) Concurrent insert of new colliding value.

In this scenario it is possible that the reader would get not found on the previous key value despite it being present in the map.